dts/bindings/interrupt-controller/nordic,split-channels.yaml

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+description: Nordic Split Channels
+
+properties:
+    owned-channels:
+      type: array
+      required: true
```

Comment:
@adamkondraciuk wouldn't that be a problem for NRF54H?

Reply:
After dropping it, when user won't define `owned-channels` property some compilation errors will appear which are not clear (related to `LISTIFY` macros expansions etc.). If you want to get rid of this requirement I would suggest to add a guard here https://github.com/zephyrproject-rtos/zephyr/blob/8bb464f24d45e30ecb31d5b508d9976817049560/modules/hal_nordic/nrfx/nrfx_config_nrf54l15_enga_application.h#L257-L261 ,
like:
```
#if DT_NODE_HAS_PROP(DT_INST(0, nordic_nrf_grtc), owned_channels)
#define _NRFX_GRTC_NUM_OF_OWNED_CHANNELS DT_PROP_LEN(DT_INST(0, \
			nordic_nrf_grtc), owned_channels)
#define _NRFX_GRTC_SHIFT_OWNED(val, _) (1 << (DT_PROP_BY_IDX(DT_INST(0, \
			nordic_nrf_grtc), owned_channels, val)))
#define _NRFX_GRTC_OWNED_CHANNEL_MASK \
			(LISTIFY(_NRFX_GRTC_NUM_OF_OWNED_CHANNELS, _NRFX_GRTC_SHIFT_OWNED, (|)))
#else
#error "property `owned-channels` does not exist.
#endif
```
@anangl - is it acceptable?

Reply:
Yes, but instead of `LISTIFY()` I'd rather use `DT_FOREACH_PROP_ELEM_SEP()`, like [here](https://github.com/zephyrproject-rtos/zephyr/blob/f58b59a4f6d84a56d94ebf61e4272d1acd0d8bd1/modules/hal_nordic/nrfx/nrfx_config.h#L669-L686).

Reply:
I' ve checked something like this:
```
#if DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_grtc)
#if DT_NODE_HAS_PROP(DT_INST(0, nordic_nrf_grtc), owned_channels)
#define NRFX_CONFIG_BIT_DT(node_id, prop, idx) \
	BIT(DT_PROP_BY_IDX(node_id, prop, idx))
#define NRFX_CONFIG_GRTC_MASK_DT(prop) \
	(COND_CODE_1(DT_NODE_HAS_PROP(DT_INST(0, nordic_nrf_grtc), prop), \
		(DT_FOREACH_PROP_ELEM_SEP(DT_INST(0, nordic_nrf_grtc), prop, \
					  NRFX_CONFIG_BIT_DT, (|))), \
		(0)))

#define NRFX_GRTC_CONFIG_ALLOWED_CC_CHANNELS_MASK \
	(NRFX_CONFIG_GRTC_MASK_DT(owned_channels) & \
	 ~NRFX_CONFIG_GRTC_MASK_DT(child_owned_channels))
#define NRFX_GRTC_CONFIG_NUM_OF_CC_CHANNELS \
	(DT_PROP_LEN_OR(DT_INST(0, nordic_nrf_grtc), owned_channels, 0) - \
	 DT_PROP_LEN_OR(DT_INST(0, nordic_nrf_grtc), child_owned_channels, 0))

#if((NRFX_CONFIG_GRTC_MASK_DT(owned_channels) | NRFX_CONFIG_GRTC_MASK_DT(child_owned_channels)) != \
		NRFX_CONFIG_GRTC_MASK_DT(owned_channels))
#error "`child-owned-channels` property must be a subset of `owned-channels` property"
#endif
#else
#error "property `owned-channels` does not exist"
#endif /* DT_NODE_HAS_PROP(DT_INST(0, nordic_nrf_grtc), owned_channels) */
#endif /* DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_grtc) */

#endif /* NRFX_GRTC_CONFIG_ALLOWED_CC_CHANNELS_MASK */
```

Reply:
It would be better to perform this error checking in e.g. `nrf_grtc_timer.c`, where it actually matters. Otherwise, the `#if` condition will be evaluated (i.e. the masks will be created) for every file that includes `nrfx.h`, so for almost all source files, what is not that great.

Reply:
@anangl I could not put @adamkondraciuk code in the `nrf_grtc_timer.c` file, because I got compilation errors (static assertion error). I put it in the `nrfx_config_` file, in place of previous definitions.

Reply:
You cannot put all this code to `nrf_grtc_timer.c` and I haven't suggested that. I mentioned only the error checking, i.e. those `#if`s that can produce `#error`s. But let's leave this as it is now. I can clean this up in #68043, where I'll need to make that mask generation a common thing for nRF54L15 and nRF54H20.